lsp-clients.el

```diff
@@ -504,7 +504,8 @@ responsiveness at the cost of possibile stability issues."
  (make-lsp-client :new-connection (lsp-stdio-connection
                                    (lambda () lsp-clients-angular-language-server-command))
                   :activation-fn (lambda (&rest _args)
-                                   (string-match-p ".*\.html$" (buffer-file-name)))
+                                   (when (file-exists-p (concat (car (project-roots (project-current))) "angular.json"))
```

Comment:
you should use f-join instead of concat and lsp-workspace-root instead of project-current since project-current does not necessarily match the project-root (e. g. if user has picked interactively project root).

Reply:
One more thing - lsp-workspace-root does not work when the project is yet to be added(e. g. you are just opening the project). In this case the angular ls wont start on the initial startup. A workaround would be the user to use lsp-workspace-folder-add before importing the project. This is a limitation of lsp-mode design - we have to first find whether there are servers matching the file and then we look for a root folder.